src/audio/audio.cpp

```diff
@@ -486,16 +498,30 @@ void Audio::doCapture()
     }
 #endif
 
-    for (size_t i = 0; i < AUDIO_FRAME_SAMPLE_COUNT * AUDIO_CHANNELS; ++i)
-        buf[i] *= inGain;
+    for (quint32 i = 0; i < AUDIO_FRAME_SAMPLE_COUNT * AUDIO_CHANNELS; ++i)
+    {
+        int16_t ampPCM = static_cast<int16_t>(qRound(buf[i] * gain));
+        buf[i] = static_cast<int16_t>(ampPCM);
+    }
 
     emit frameAvailable(buf, AUDIO_FRAME_SAMPLE_COUNT, AUDIO_CHANNELS, AUDIO_SAMPLE_RATE);
 }
 
 /**
+@internal
+*/
+void Audio::privateSetGain(qreal percentage)
+{
+    percentage = std::max(0.0, std::min(percentage, 1.0));
+    gainPercentage = percentage;
+
+    gain = (percentage > 0.0) ? 10.0 * qAbs(log10(percentage * 100.0)) : 0.0;
```

Comment:
The use of `qAbs()` in this context causes incorrect behaviour when input gain percentage is < 1% (it causes a massive rise in gain since the log curve is reflected upwards).